python-frontend/src/main/java/org/sonar/python/semantic/FunctionSymbolImpl.java

```diff
@@ -107,18 +111,19 @@ FunctionSymbolImpl copyWithoutUsages() {
 
   private static boolean isInstanceMethod(FunctionDef functionDef) {
     return !functionDef.name().name().equals("__new__") && functionDef.isMethodDefinition() && functionDef.decorators().stream()
-      .map(decorator -> {
-        List<Name> names = decorator.name().names();
-        return names.get(names.size() - 1).name();
-      })
+      .map(decorator -> TreeUtils.decoratorNameFromExpression(decorator.expression()))
+      .filter(Objects::nonNull)
+      .flatMap(s -> Arrays.stream(s.split("\\.")))
```

Comment:
I think we don't need this `flatMap` . We want to check that decorator name is equal to "staticmethod" or "classmethod"

Reply:
Indeed, thanks! I thought you could use them with another qualified name like `abc.staticmethod` hence why I kept it. But after double checking it I was mistaken.